tests/test_board.rb

```diff
@@ -3,31 +3,101 @@
 require_relative '../src/player'
 
 describe Board do
+    it 'should verify if the game has a valid board of 8 squares' do
+        board = Board.new
+
+        valid = board.isBoardValid?
+        
+        assert valid
+    end
+
     it 'should provide access to the board' do
         board = Board.new
         boardArray = board.board
+
         assert_equal (0..8).to_a, boardArray
     end
 
     it 'should allow for the board to be set to an array' do
        board = Board.new
        array = [0,1,2]
        board.board = array
+
        assert_equal array, board.board
     end
 
+    it 'should check to see if there are empty spots on the board' do
+        board = Board.new
+        board.board = [0, 1, 2, 3, 4, 5, 6, 'X', 8]
+        
+        isFull = board.isFull?
+        
+        assert_equal false, isFull
+    end
+
+    it 'should return an array with the index number of all the positions of the given player' do
```

Comment:
what about if there's other player's pieces on the board? what is the board has no player pieces?